examples/chat_with_your_documents/main.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import os
+from typing import List, Dict
+
+import openai
+import chromadb
+
+
+def build_prompt(query: str, context: List[str]) -> List[Dict[str, str]]:
+    """
+    Builds a prompt for the LLM. #
+
+    This function builds a prompt for the LLM. It takes the original query,
+    and the returned context, and asks the model to answer the question based only
+    on what's in the context, not what's in its weights.
+
+    More information: https://platform.openai.com/docs/guides/chat/introduction
+
+    Args:
+    query (str): The original query.
+    context (List[str]): The context of the query, returned by embedding search.
+
+    Returns:
+    A prompt for the LLM (List[Dict[str, str]]).
+    """
+
+    system = {
+        "role": "system",
+        "content": "I am going to ask you a question, which I would like you to answer"
+        "based only on the provided context, and not any other information."
+        "If there is not enough information in the context to answer the question,"
+        'say "I am not sure", then try to make a guess.'
+        "Break your answer up into nicely readable paragraphs.",
+    }
+    user = {
+        "role": "user",
+        "content": f"The question is {query}. Here is all the context you have:"
+        f'{(" ").join(context)}',
+    }
+
+    return [system, user]
+
+
+def get_chatGPT_response(query: str, context: List[str]) -> str:
+    """
+    Queries the GPT API to get a response to the question.
+
+    Args:
+    query (str): The original query.
+    context (List[str]): The context of the query, returned by embedding search.
+
+    Returns:
+    A response to the question.
+    """
+
+    response = openai.ChatCompletion.create(
+        model="gpt-3.5-turbo",
+        messages=build_prompt(query, context),
+    )
+
+    return response.choices[0].message.content  # type: ignore
+
+
+def main(
+    collection_name: str = "documents_collection", persist_directory: str = "."
+) -> None:
+    # Check if the OPENAI_API_KEY environment variable is set. Prompt the user to set it if not.
+    if "OPENAI_API_KEY" not in os.environ:
+        openai.api_key = input(
+            "Please enter your OpenAI API Key. You can get it from https://platform.openai.com/account/api-keys\n"
+        )
+
+    # Instantiate a persistent chroma client in the persist_directory.
+    # This will automatically load any previously saved collections.
+    # Learn more at docs.trychroma.com
+    client = chromadb.PersistentClient(path=persist_directory)
+
+    # Get the collection.
+    collection = client.get_collection(name=collection_name)
```

Comment:
doesnt use openai here either

Reply:
This file uses `openai` to call out to the LLM.